docs/topics/data-analysis/data-analysis-work-with-api.md

```diff
@@ -1,18 +1,201 @@
+unnecessary ones for clarity:
+
+```kotlin
+val videos = items.dropNulls { id.videoId }
+    .select { id.videoId named "id" and snippet }
+    .distinct()
+videos
+```
+
+2. **Chunk and load data:**
+Chunk IDs from the cleaned data and load corresponding video statistics. This involves breaking the data into smaller 
+batches and fetching additional details:
+
+```kotlin
+val statPages = clean.id.chunked(50).map {
+    val ids = it.joinToString("%2C")
+    load("videos?part=statistics&id=$ids")
+}
+statPages
+```
+
+3. **Concatenate and select relevant data:**
+Concatenate the fetched statistics and select relevant columns:
+
+```kotlin
+val stats = statPages.items.concat().select { id and statistics.all() }.parse()
+stats
+```
+
+4. **Join DataFrames:**
+Join the existing cleaned data with the newly fetched statistics. This merges two sets of data into a comprehensive DataFrame:
+
+```kotlin
+val joined = clean.join(stats)
+joined
+```
+
+This example demonstrates how to clean, reorganize, and enhance your dataset using Kotlin DataFrame's various functions. 
+Each step is designed to refine the data, making it more suitable for [in-depth analysis](#analyze-data-in-kotlin-notebook).
+
+## Analyze data in Kotlin Notebook
+
+After you've successfully [fetched](#fetch-data-from-an-api) and [cleaned and refined your data](#clean-and-refine-data) 
+using functions from the [Kotlin DataFrame library](https://kotlin.github.io/dataframe/gettingstarted.html), the next step 
+is to analyze this prepared dataset to extract meaningful insights.
+
+Methods such as [`.groupBy`](https://kotlin.github.io/dataframe/groupby.html) for categorizing data, 
+[`.sum`](https://kotlin.github.io/dataframe/sum.html) and [`.maxBy`](https://kotlin.github.io/dataframe/maxby.html) for 
+[summary statistics](https://kotlin.github.io/dataframe/summarystatistics.html), and [`.sortBy`](https://kotlin.github.io/dataframe/sortby.html) for ordering data are particularly useful. 
+These tools allow you to perform complex data analysis tasks efficiently. 
+
+Let's look at an example, using `.groupBy` to categorize videos by channel, `.sum` to calculate total views per category, 
+and `.maxBy` to find the latest or most viewed video in each group:
+
+1. **Set up column references:** Simplify the access to specific columns by setting up references:
+
+```kotlin
+val view by column<Int>()
+```
+
+2. **Group and sort data:** Use the `.groupBy` method to group the data by the `channel` column and sort it. 
+
+```kotlin
+val channels = joined.groupBy { channel }.sortByCount()
+```
+
+In the resulting table, you can interactively explore the data. Clicking on the `group` field 
+of a row corresponding to a channel will expand that row to reveal more details about that channel's videos.
+
+You can click on the table icon in the bottom left to return to the grouped dataset.
+
+![Expanding a row](results-of-expanding-group-data-analysis.png){width=700}
```

Comment:
These images are not visible. Consider adding one at a time with a leading line. For example, "This is how the grouped data looks like: [IMAGE 1] . This is how the data looks like after clicking on a row [IMAGE 2]"

Reply:
I agree - and that's a nice suggestion I'll for with that one! :)